cfgov/v1/tests/models/test_related_posts.py

```diff
@@ -148,6 +149,25 @@ def test_related_posts_blog_limit(self):
         self.assertNotIn('Newsroom', related_posts)
         self.assertNotIn('Events', related_posts)
 
+    def test_related_posts_and_filtering(self):
+        """
+        Tests whether related posts are retrieved if and only if they match ALL
+        of the tags on the calling page.
+        """
+
+        self.block.value['relate_posts'] = True
+        self.block.value['relate_newsroom'] = True
+        self.block.value['relate_events'] = True
```

Comment:
@richaagarwal It was intended to demonstrate that the single-tagged posts found in `posts` and `events` are available, but not retrieved when `and_filtering` is `True`.

I think @chosak's solution of creating a new test explicitly testing the `False` condition for `and_filtering` is a good idea. I could also add more detailed comments, if that would help, @richaagarwal.